apps/wallet-mobile/src/features/WalletManager/context/WalletManagerProvider.tsx

```diff
@@ -84,13 +84,19 @@ export const WalletManagerProvider: React.FC<
 }
 
 export const useWalletManager = () => {
-  const context = React.useContext(WalletManagerContext)
+  const {selected, walletManager} = React.useContext(WalletManagerContext)
 
-  if (context.walletManager == null) {
-    const error = new Error('useWalletManager wallet manager is not set, invalid state reached')
+  if (walletManager == null) {
+    const error = new App.Errors.InvalidState('useWalletManager wallet manager is not set, invalid state reached')
     logger.error(error)
     throw error
   }
 
-  return context
+  return React.useMemo(
```

Comment:
@michaeljscript otherwise TS doesn't infer that that walletManager is not null, had to expand it.